src/Bladeburner/Bladeburner.ts

```diff
@@ -1156,10 +1156,7 @@ export class Bladeburner {
           }
           case BladeGeneralActionName.diplomacy: {
             const eff = this.getDiplomacyEffectiveness(person);
-            this.getCurrentCity().chaos *= eff;
-            if (this.getCurrentCity().chaos < 0) {
-              this.getCurrentCity().chaos = 0;
-            }
+            this.getCurrentCity().changeChaosByPercentage((1 - eff) * 100);
```

Comment:
`eff` is provided as a decimal multiplier intended to be multiplied against the chaos, e.g. `0.98` to apply a 2% change. However, changeChaosByPercentage wants the percentage, not the decimal. So, (1 - eff) * 100 converts the decimal multiplier to the percentage differential.

Reply:
This one is complicated. For both numerical reasons (numbers near zero have precision in floating-point, which is lost if you add/subtract them from one), and also because it makes the calculation much harder to understand, I think this one should be kept as a straight multiply, and then use `clampNumber(chaos, 0)` after.

Reply:
Actually, hold on one second....

Reply:
OK, I have another option that might be better. I think getDiplomacyEffectiveness is only used here, in this one place. Also, it internally is calculating a percentage and then converting it to effectiveness. So you could change the function to getDiplomacyPercentage and just have it return the percentage directly, and then feed it directly to changeChaosByPercentage.

If you don't want to do all that, the original option stands. Doing more means doing more verification work.

Reply:
I like the idea of changing it! I'll put that together.

I really don't want the attribute being manually adjusted outside of the intended functions, personally.

Reply:
Implemented!